frontend/gatsby-node.js

```diff
@@ -7,12 +7,367 @@
+        .replace(/\./g, "")
+        .replace(/\s+/g, "-"),
+    })
+  );
+  // All Series List for Drop Down Selection
+  let series = resultSeries?.data?.allStrapiSermonSeries?.nodes.map(
+    ({ Name }) => ({
+      label: Name,
+      value: Name.toLowerCase()
+        .replace(/[.'"&]./g, "")
+        .replace(/\s+/g, "-"),
+    })
+  );
+  // All Bible Books List for Drop Down Selection
+  let books = posts.map(({ BiblePassage }) =>
+    BiblePassage.map(({ Book }) => Book)
+  );
+  books = [].concat(...books).map(book => {
+    const bookLabel = /\(\d[a-z]+\)/.test(book)
+      ? `${book.substring(
+          book.search(/\d/),
+          book.search(/\d/) + 1
+        )} ${book.substring(0, book.search(/\(/))}`
+      : book;
+    return {
+      label: bookLabel,
+      value: book.replace(/\(|\)/g, "").replace(/\s+/, "-").toLowerCase(),
+      book,
+    };
+  });
+
+  const postsPerPage = ITEMS_PER_PAGE;
+  const numberOfPages = Math.ceil(posts.length / postsPerPage);
+  // Update Speaker List to include the number of pages of sermons for each speaker and filter the list to create a list of speakers that has at least one sermon.
+  speakers = speakers
+    .map(speaker => ({
+      ...speaker,
+      numOfPages: Math.ceil(
+        posts.filter(
+          ({ Preacher: { Prefix, Name } }) =>
+            speaker.prefix === Prefix && speaker.name === Name
+        ).length / postsPerPage
+      ),
+    }))
+    .filter(({ numOfPages }) => numOfPages > 0);
+
+  // Update Series List to include number of pages of sermons for each series.
+  series = series.map(item => ({
+    ...item,
+    numOfPages: Math.ceil(
+      posts.filter(({ Series: { Name } }) => item.label === Name).length /
+        postsPerPage
+    ),
+  }));
+  // Update Bible Books List to include number of pages of sermons for each Bible Book.
+  books = books.map(item => ({
+    ...item,
+    numOfPages: Math.ceil(
+      posts.filter(({ BiblePassage }) =>
+        BiblePassage.some(({ Book }) => Book === item.book)
+      ).length / postsPerPage
+    ),
+  }));
+  // Create Pages for all sermons and pass data in the PageContext Object.
+  Array.from({ length: numberOfPages }).forEach((_, index) => {
+    createPage({
+      path: index === 0 ? `/watch` : `/watch/${index + 1}`,
+      component: path.resolve("./src/templates/sermons.js"),
+      context: {
+        limit: postsPerPage,
+        skip: index * postsPerPage,
+        numPages: numberOfPages,
+        currentPage: index + 1,
+        speakers,
+        series,
+        books,
+      },
+    });
+  });
+  /* Loop through speakers to create pages for one of the follow combinations:
+   * Speaker's Sermon, Speaker's And Specific Series' Sermons, and Speaker's, Specific Series' and Specific Bible Book's Sermons
+   */
+  speakers.forEach(({ value, prefix, name, numOfPages }) => {
+    // Create pages for Each Speaker with their sermons.
+    Array.from({ length: numOfPages }).forEach((_, index) => {
```

Comment:
This is working at the moment, but this is 2^n for number of categories we have lol. In the future if we need another category, for example SermonTopics, then we'll need to put in another 8 createPage calls... probably need to switch over to some kind of  recursive call at some point.